src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+        var moved = false
+        var x2 = 0
+        var y2 = 0
+        while i < |desired| && moved == false do
+          val x = desired(i)
+          val y = desired(i + 1)
+          let
+            up = y > this.ymax
+            down = y < this.ymin
+            right = x > this.xmax
+            left = x < this.xmin
+            border = if up then
+                       if right then
+                         this.top_right
+                       else
+                         if left then
+                           this.top_left
+                         else
+                           this.up
+                         end
+                       end
+                     else
+                       if down then
+                         if right then
+                           this.bottom_right
+                         else
+                           if left then
+                             this.bottom_left
+                           else
+                             this.down
+                           end
+                         end
+                       else
+                         if left then
+                           this.left
+                         else
+                           if right then
+                             this.right
+                           else
+                             Nothing
+                           end
+                         end
+                       end
+                     end
+          in
+              var ax = a.a.x
+              var ay = a.a.y
+              match border with
+                case Nothing =>
+                  if not(y > this.ymax || y < this.ymin || x > this.xmax || x < this.xmin) && this.matrix.get_val(x, y) == false then
+                    this.matrix.remove(ax, ay)
+                    this.matrix.set(x, y)
+                    a.a.move_int(x, y)
+                    moved = true
+                  end
+                end
+                case Just(real_thing) =>
+                    x2 = ax
+                    y2 = ay
+                    val first= (wrapped_a == this.agents)
+                    var nejbor = real_thing ! external_move( a.a, x, y) --consume
+                    await(nejbor)
+                    val futurefilled = get(nejbor)
+                    match futurefilled with
+                        case Just(returned_agent) => a.a=returned_agent
+                        case Nothing =>
+                              this.size = this.size - 1
+                              this.matrix.remove(x2, y2)
+                              moved = true
+                              if first then
+                                 this.agents = a.next
+                              end
+                        end
+                    end
+                end
+              end
+              i = i + 2
+          end
+        end
+      end
+      wrapped_a = a.next
+    end
+    this.moving = false
+    true
```

Comment:
Changed to unit. I won't re-factor, not worth the effort as verifying that this method is correct is difficult, and I have tested it as it looks now.